compiler/tests/fixtures/data_copying_stress.fe

```diff
@@ -0,0 +1,57 @@
+contract Foo:
+    my_string: string42
+    my_other_string: string42
+
+    my_u256: u256
+    my_other_u256: u256
+
+    my_nums: u256[5]
+
+    event MyEvent:
+        my_string: string42
+        my_u256: u256
+
+    pub def set_my_vals(
+        my_string: string42,
+        my_other_string: string42,
+        my_u256: u256,
+        my_other_u256: u256
+    ):
+        self.my_string = my_string
+        self.my_other_string = my_other_string
+        self.my_u256 = my_u256
+        self.my_other_u256 = my_other_u256
+
+    pub def set_to_my_other_vals():
+        self.my_string = self.my_other_string
+        self.my_u256 = self.my_other_u256
+
+    pub def mutate_and_return(my_array: u256[10]) -> u256[10]:
+        my_array[3] = 5
+        return my_array
+
+    pub def clone_and_return(my_array: u256[10]) -> u256[10]:
+        return my_array.clone()
+
+    pub def clone_mutate_and_return(my_array: u256[10]) -> u256[10]:
+        my_array.clone()[3] = 5
+        return my_array
+
+    pub def assign_my_nums_and_return() -> u256[5]:
+        my_nums_mem: u256[5]
+        self.my_nums[0] = 42
+        self.my_nums[1] = 26
+        self.my_nums[2] = 0
+        self.my_nums[3] = 1
+        self.my_nums[4] = 255
+        my_nums_mem = self.my_nums.to_mem()
+        return my_nums_mem
+
+    pub def emit_my_event():
+        self.emit_my_event_internal(
+            self.my_string.to_mem(),
+            self.my_u256.to_mem()
```

Comment:
So, now spotted that further down in `sized_vals_in_sto.fe` we have this:

```
 emit MyEvent(
            self.num,
            self.nums.to_mem(),
            self.str.to_mem()
        ) 
```

Here we don't call `.to_mem()` on `self.num` which is what I would expect. So, I'm assuming that maybe this is a bug?